File: mobile/src/main/java/treehou/se/habit/main/MainActivityModule.java
package treehou.se.habit.main;


import dagger.Module;
import dagger.Provides;
import treehou.se.habit.module.ViewModule;

@Module
public class MainActivityModule extends ViewModule<MainActivity> {
    public MainActivityModule(MainActivity activity) {
        super(activity);
    }

    @Provides
    public MainContract.View provideView() {
        return view;
    }
}

File: mobile/src/main/java/treehou/se/habit/main/MainActivityModule.kt
package treehou.se.habit.main


import dagger.Module
import dagger.Provides
import treehou.se.habit.module.ViewModule

@Module
class MainActivityModule(activity: MainActivity) : ViewModule<MainActivity>(activity) {

    @Provides
    fun provideView(): MainContract.View {
        return view
    }
}
